### PR TITLE
Inputdriver ui improvement

### PR DESCRIPTION
**Pull request by MichaelPFrey**
- bar changes color when above deadzone value
- output status information in the axis config dialog
- implementation details improved
- making better use of the status information provided by qgamepad

### CODE REVIEW

**Comment by MichaelPFrey** [comment]
bildschirmfoto von 2019-02-25 06-39-12

File: src/input/AxisConfigWidget.cc [hunk 1]
```diff
@@ -51,11 +51,24 @@ void AxisConfigWidget::AxesChanged(int nr, double val) const{
 	if(progressBar==nullptr) return;
 	progressBar->setValue(value);
 	progressBar->setFormat(s);
+	
+	QDoubleSpinBox* deadzone = this->findChild<QDoubleSpinBox *>(QString::fromStdString("doubleSpinBoxDeadzone"+number));
```
Comment (kintel) [inline src/input/AxisConfigWidget.cc]: Tiny suggestion: You can concatenate numbers directly to a QString: `QString("doubleSpinBoxDeadzone") + nr`, without going through a std::string.

File: src/input/AxisConfigWidget.cc [hunk 2]
```diff
@@ -51,11 +51,24 @@ void AxisConfigWidget::AxesChanged(int nr, double val) const{
 	if(progressBar==nullptr) return;
 	progressBar->setValue(value);
 	progressBar->setFormat(s);
+
```
Comment (kintel) [inline src/input/AxisConfigWidget.cc]: I didn't catch this earlier, but `setFormat()` accepts a format string, not a number. Not sure if your way triggers some undocumented compatibility mode perhaps?

File: src/input/DBusInputDriver.cc [hunk 3]
```diff
@@ -150,3 +150,10 @@ const std::string & DBusInputDriver::get_name() const
 {
     return name;
 }
+
+std::string DBusInputDriver::get_info() const
+{
+	std::ostringstream stream;
+	stream << get_name() << " " << (isOpen() ? "open" : "not open");
+	return stream.str();
```
Comment (kintel) [inline src/input/DBusInputDriver.cc]: You can replace this by `return STR(get_name() << " " << (isOpen() ? "open" : "not open"))`

-> We have a STR macro for this common operation: https://github.com/openscad/openscad/blob/master/src/printutils.h#L100

File: src/input/AxisConfigWidget.cc (hunk 1 above)
Comment (MichaelPFrey) [reply src/input/AxisConfigWidget.cc]: that makes things simpler - thank you

File: src/input/AxisConfigWidget.cc (hunk 2 above)
Comment (MichaelPFrey) [reply src/input/AxisConfigWidget.cc]: Hmm... that needs documentation from my side.

QProgressbar does not have a way of setting a text directly.
All I can do is setting a format, that is used to create the text.
By setting a format that does not contain a placeholder for the value, only the format string is shown.
By using a format string that is the way I want the string to be formatted, I get the result I want.

The alternative is, to overlay the progressbar with a label. Sounds messy.

Alternatively, I could look at QSlider or QScrollbar, but they do not have any text. (where I would like to have it)

Or customize the QProgressbar.

**Comment by MichaelPFrey** [comment]
axiswindowsspaceexplorer

**Comment by MichaelPFrey** [comment]
This good for review for now, otherwise I will increase the scope more and more.

**Comment by kintel** [comment]
Ok, will review. Limited scope is good :)

**Pull request merged**